source/dagger/gameplay/platformer/platformer_controller.cpp

```diff
@@ -26,19 +29,204 @@ void PlatformerControllerSystem::SpinUp()
 
 void PlatformerControllerSystem::Run()
 {
-    Engine::Registry().view<InputReceiver, Sprite, Animator, PlatformerCharacter>().each(
-        [](const InputReceiver input_, Sprite& sprite_, Animator& animator_, const PlatformerCharacter& char_)
+    Engine::Registry().view<InputReceiver, Sprite,Transform, Animator, PlatformerCharacter>().each(
+        [](const InputReceiver input_, Sprite& sprite_, Transform& transform_,Animator& animator_,PlatformerCharacter& char_)
```

Comment:
might be useful to write smaller functions and call them here. But wait for today's lecture, we will discuss similar thing today